### DISCUSSION TITLE
Status 400 received when requesting transcript

### DISCUSSION THREAD

**Discussion by beirne** [lstrzepek/obsidian-yt-transcript#20] [Q&A] [answered]
I just installed the plugin into Obsidian on my Mac but it can't pull transcripts. I get a status 400 error. This happens with videos that have both auto-generated and creator-generated transcripts.

<img width="260" alt="image" src="https://github.com/lstrzepek/obsidian-yt-transcript/assets/6078590/50905047-6596-4239-b14b-e7b56e0c5349">

I'm running YTranscript 0.7.1 on Obsidian 1.3.5 on MacOS 13.4.

**Comment by decaf-dev** [comment]
@beirne Try v0.8.1

**Reply by beirne** [reply]
I'm still getting the 400 status with 0.8.1. I tried both ways of downloading the transcript and got the same result with both.

**Reply by decaf-dev** [reply]
That's odd. I don't get that error in 0.8.1. What kind of YouTube links are you using?

**Reply by beirne** [reply]
Here's one: https://youtu.be/FatnXnlwAc4

**Reply by decaf-dev** [reply, marked as answer]
Ah. I get the 400 error as well. Try https://www.youtube.com/watch?v=FatnXnlwAc4. This is the full youtube URL. It includes the `watch?v=` and `youtube.com` instead of `youtu.be`. It appears that `youtu.be` links are broken.

**Reply by beirne** [reply]
Yep, that's it. Thanks!

**Reply by decaf-dev** [reply]
I have found the problem in the code. I might push a fix.

**Reply by Kamilkopec** [reply]
Hi!
Have this issue while on iPhone 12 mini using the Yt app. All up to date including iOS.
Reproduction:
- open YT app 
- click on selected movie 
- click share 
![image](https://github.com/lstrzepek/obsidian-yt-transcript/assets/35232085/b181598f-c266-4a1b-b133-0ade5582591f)

- I'm getting this url: https://youtu.be/Yt3JJ-e7g9g?si=R37f_BQHTVvFXpPC
It does not work.

Another option:
I click share -> more -> copy 
And I get the following link:

https://youtube.com/watch?v=Yt3JJ-e7g9g&si=R37f_BQHTVvFXpPC

Also does not work,
Tried multiple movies same issue.

**Reply by lstrzepek** [reply]
Thank you very much for helping other solve problems. Hope now all above urls are working as expected

**Reply by NJonasFigge** [reply]
To anyone facing this issue on a toolbar created with the "Note toolbar" plugin:
- Go to the item in the toolbar settings
- Click on "Advanced settings" (gear icon)
- Enable "Focus in editor"

Context: Although executing YTranscript on selected url from command palette works fine, the toolbars seem to behave differently, leading to the above error. I know that this is not an issue and not really connected to this repo, but I landed here, before figuring it out...

Hope at least one person is saved the trouble :) Have a nice one!
